### PR TITLE
add cjs support for machine-check

### PR DESCRIPTION
**Pull request by Kelerchian**

### CODE REVIEW

File: machine-check/examples/esm/example.test.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ const subscriptions = {
 
 describe('swarmProtocol', () => {
   it('should be well-formed', () => {
-    expect(checkSwarmProtocol(swarmProtocol, subscriptions)).toEqual({ type: 'OK' })
+    expect(checkSwarmProtocol(swarmProtocol, subscriptions)).not.toEqual({ type: 'OK' })
```
Comment (rkuhn) [inline machine-check/examples/esm/example.test.ts:52]: I’d rather check that the errors are as expected and/or fixing the protocol to make it well-formed.

File: machine-check/scripts/test/cjs-copy.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// ${sourcePath} 
+`.trim() + '\n\n'
+
+globSync(TEST_ESM_GLOB, {
+  ignore: ['**/jest.config.ts'],
+}).forEach((sourcePath) => {
+  const destinationPath = path.resolve(TEST_CJS_DIR, path.relative(TEST_ESM_DIR, sourcePath))
+
+  let content = fs.readFileSync(sourcePath, 'utf8')
+  content = content.replace(/from '(.*).js'/gi, (_, p1) => `from '${p1}'`)
```
Comment (rkuhn) [inline machine-check/scripts/test/cjs-copy.ts:18]: We don’t have dynamic imports, right? (I just remembered that this was the problem with konva, so we might also patch `import('...')` syntax proactively) Or is it guaranteed that running the tests will catch such an issue?

**Review by rkuhn — APPROVED** [review]
Looking good! Feel free to fix the well-formedness test issue either way (and ask if unsure)

File: machine-check/examples/esm/example.test.ts (hunk 1 above)
Comment (Kelerchian) [reply machine-check/examples/esm/example.test.ts:52]: Should we create cases for when the SwarmProtocol is well-formed?

File: machine-check/scripts/test/cjs-copy.ts (hunk 2 above)
Comment (rkuhn) [reply machine-check/scripts/test/cjs-copy.ts:18]: https://github.com/konvajs/konva/pull/1514/files

**Review by rkuhn — APPROVED** [review]

File: machine-check/examples/esm/example.test.ts (hunk 1 above)
Comment (rkuhn) [reply machine-check/examples/esm/example.test.ts]: yes, thanks!

**Pull request merged**